cvat-core/src/annotations-actions.ts

```diff
@@ -82,6 +95,45 @@ class RemoveFilteredShapes extends BaseSingleFrameAction {
     }
 }
 
+class PropagateShapes extends BaseSingleFrameAction {
+    #targetFrame: number;
+
+    public async init(instance, parameters): Promise<void> {
+        this.#targetFrame = parameters['Target frame'];
+    }
+
+    public async destroy(): Promise<void> {
+        // nothing to destroy
+    }
+
+    public async run(
+        instance,
+        { collection: frameCollection, frameData: { number } },
+    ): Promise<SingleFrameActionOutput> {
+        const { shapes } = frameCollection;
+        const propagatedShapes = propagateShapes(shapes, number, this.#targetFrame);
+        return { collection: { shapes: [...shapes, ...propagatedShapes] } };
+    }
+
+    public get name(): string {
+        return 'Propagate shapes';
+    }
+
+    public get parameters(): ActionParameters | null {
+        return {
+            'Target frame': {
+                type: ActionParameterType.NUMBER,
+                values: ['0', 'frameCount', '1'],
```

Comment:
I like the idea of using "special values", but probably we should be more specific here.
So, I will propose to use callback, accepting an object with job and returning values, e.g.:

`({ job }) => job.startFrame` instead of `'0'` and `({ job }) => job.stopFrame` instead of `'frameCount'`

Additionally converted to strings. 

Also, frame count is 12 in a job with 11 frames, looks like a small issue. 
This approach will be quite general. 

Later we may add more input data to the callback, supporting backward compatibility

Reply:
Or instead of `job`, maybe to use `instance: Job | Task` as actions are already implemented for both. 
Need to think. .

Reply:
As we dont really run actions on tasks right now, I suppose `Job` would be enough

Reply:
But if we will replace it by `instance` in the future, it will break backward compatibility